src/arrayMethodSort.js

```diff
@@ -4,8 +4,38 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = null) {
+    if (this.length < 2) {
+      return this;
+    }
+
+    let compareFn = compareFunction;
+
+    if (typeof compareFunction !== 'function') {
+      compareFn = (a, b) => {
+        if (String(a) < String(b)) {
+          return -1;
+        }
+
+        if (String(a) > String(b)) {
+          return 1;
+        }
+
+        return 0;
+      };
```

Comment:
let's move your default fn outside your method and assign it as the default value for the parameter. In this case you can avoid checking type.